financial-connections/src/main/java/com/stripe/android/financialconnections/features/partnerauth/SharedPartnerAuthState.kt

```diff
@@ -48,7 +46,8 @@ internal data class SharedPartnerAuthState(
             authenticationStatus !is Loading &&
                 authenticationStatus !is Success &&
                 // Failures posting institution -> don't allow back navigation
-                payload !is Fail
+                payload !is Fail &&
+                !isNetworkingRelinkSession
```

Comment:
nit - maybe a comment to clarify why back navigation is not allowed here would help people with no repair context